### PR TITLE
ip1 instructions

### PR DESCRIPTION
**Pull request by iamabhutta**
ip1 instructions. Just need to link the code to gitclassroom. Rest is ready to go.

### CODE REVIEW

File: Assignments/ip1/ip1.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+3. Start by defining the schemas in the server/models/schema directory to ensure the data structure is consistent.
+4. Ensure that you run all Jest tests. These tests are designed to catch issues early. Once all Jest tests pass, the Cypress tests should also pass, assuming no significant changes have been made to the client’s implementation.
+5. Use Postman to interact with and manually test your solutions and verify that database queries work as expected.
+6. Follow the [debugging policy](https://neu-se.github.io/CS4530-Spring-2025/policies/debugging/) to help in the debugging process.
+
+## Implementation Tasks
+
+This deliverable has 2 parts; each part will be graded on its own rubric. You should complete the assignment one part at a time, in the order presented here.
+
+### Task 1: Add `User`s
```
Comment (VatsalMehta27) [inline Assignments/ip1/ip1.md:213]: This shows up weird on the website as "AddUsers" -> remove the backticks surrounding it?

File: Assignments/ip1/ip1.md [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+
+   Define the mongoose schema for a `User` in `server/models/schema/user.schema.ts`. Each `User` should have a unique username, a password, and the date the account was created.
+
+2. **Define the `User` model**
+
+   In `server/models/user.model.ts`, define the mongoose model to be able to perform database operations.
+
+3. **Define relevant TypeScript types**
+
+   In `server/types/user.d.ts`, define the relevant `User` and `UserCredentials` types to mirror the schema definition. Make sure to avoid repeated definitions of fields between the interfaces.
```
Comment (VatsalMehta27) [inline Assignments/ip1/ip1.md:237]: Change to "define the relevant types marked by \_TODO: Task 1\_. Make ..."

File: Assignments/ip1/ip1.md [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+
+14. **Add the routes to the router**
+
+    Define the appropriate API routes on the Express router for each of the functions implemented in the previous steps.
+
+15. **Test all added functions**
+
+    We've provided initial tests to provide some information on the expected behavior of the routes and functions. Using the requirement descriptions above, write additional tests for all the added functions and routes, covering different branches, edge cases, etc. to verify the correctness of your code.
+
+    In addition to automated tests, you should also manually test your route using Postman and MongoDBCompass to ensure that any database queries are correct since we use database mocks while testing with Jest.
```
Comment (VatsalMehta27) [inline Assignments/ip1/ip1.md:287]: MongoDB Compass (missing space)

File: Assignments/ip1/ip1.md [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+
+   In `server/controller/message.controller.ts`, implement the `addMessageRoute` endpoint. This route will handle POST requests to create a new message with the provided parameters in the request body.
+
+6. **Implement the `getMessages` function**
+
+   In `server/services/message.service.ts`, implement the function for returning all messages from the database, sorted by date in ascending order.
+
+7. **Implement the `getMessagesRoute` endpoint**
+
+   In `server/controller/message.controller.ts`, complete the `getMessagesRoute` endpoint so that it returns all messages.
```
Comment (VatsalMehta27) [inline Assignments/ip1/ip1.md:331]: Replace path with: \`server/controllers/message.controller.ts\` (missing 's')

File: Assignments/ip1/ip1.md [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+
+   In `server/types/message.d.ts`, define the relevant `Message` type to reflect the schema defintion.
+
+4. **Implement the `saveMessage` function**
+
+   In `server/services/message.service.ts`, implement the logic for creating a new `Message` document in the database.
+
+5. **Implement the `addMessageRoute` endpoint**
+
+   In `server/controller/message.controller.ts`, implement the `addMessageRoute` endpoint. This route will handle POST requests to create a new message with the provided parameters in the request body.
```
Comment (VatsalMehta27) [inline Assignments/ip1/ip1.md:323]: Replace path with: \`server/controllers/message.controller.ts\` (missing 's')

File: Assignments/ip1/ip1.md [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+
+7. **Implement the `getMessagesRoute` endpoint**
+
+   In `server/controller/message.controller.ts`, complete the `getMessagesRoute` endpoint so that it returns all messages.
+
+8. **Test all added functions**
+
+   The frontend code of Task 2 is provided to you. If you have done everything correctly, you should now be able to run the webpage and test the functionality of the Global Chat feature manually.
+
+   In addition, write additional automated tests in `server\tests\controllers\message.controller.spec.ts` and `server\tests\services\message.service.spec.ts`.
```
Comment (VatsalMehta27) [inline Assignments/ip1/ip1.md:337]: Replace '\' with '/' for consistency.
